app/src/pages/OnDeviceDisplay/ConnectViaWifi.tsx

```
@@ -0,0 +1,184 @@
+          setChangeState={setChangeState}
+        />
+      )
+      // This condition might be changed for manual connect
+    } else if (changeState.ssid != null) {
+      if (currentRequestState === null) {
+        return (
+          <SetWifiCred
+            ssid={changeState.ssid}
+            setShowSelectAuthenticationType={setShowSelectAuthenticationType}
+            authType={selectedAuthType}
+            password={password}
+            setPassword={setPassword}
+            handleConnect={handleConnect}
+          />
+        )
+      } else if (currentRequestState.status === RobotApi.PENDING) {
+        return <ConnectingNetwork />
+      } else if (currentRequestState.status === RobotApi.SUCCESS) {
+        return (
+          <SucceededToConnect
+            ssid={changeState.ssid}
+            authType={selectedAuthType}
+          />
+        )
+      } else {
+        return (
+          <FailedToConnect
+            ssid={changeState.ssid}
+            requestState={currentRequestState}
+            type={changeState.type}
+            onConnect={handleConnect}
+            setChangeState={setChangeState}
+            setCurrentRequestState={setCurrentRequestState}
+          />
+        )
+      }
+    } else {
+      console.log('here')
+      return null
+    }
+  }
+
+  React.useEffect(() => {
+    dispatch(Networking.fetchWifiList(robotName))
+  }, [dispatch, robotName])
+
+  useInterval(
+    () => dispatch(Networking.fetchWifiList(robotName)),
+    LIST_REFRESH_MS,
+    true
+  )
+
+  React.useEffect(() => {
+    if (list != null && list.length > 0) {
+      setIsSearching(false)
+    }
+  }, [list])
+
+  React.useEffect(() => {
+    setCurrentRequestState(requestState)
+  }, [requestState])
+
+  React.useEffect(() => {
+    // a user selects none as authType
+    if (selectedSsid !== '' && selectedAuthType === 'none') {
+      const network = list.find((nw: WifiNetwork) => nw.ssid === selectedSsid)
```

i think this useEffect is causing issues on the `SelectAuthenticationType` screen - if i select "None" the connect request is made immediately, and if it fails, it fails silently. i don't see the failure until clicking "Next". instead, it seems more in line with what i expect to make the request on clicking "Next".

Related, if i select "None" then switch back to "WPA2 Personal" and click "Next", I get the failure screen without being prompted to enter a password. this is also unexpected.

instead of fixing now, we could make a bug ticket in order to get this large PR merged.

---

Will make a ticket for this